### PR TITLE
Laboratory Work #3 Natalia Antonova 17FPL-2

### PR DESCRIPTION
**Pull request by NatalieAntoshka**

### CODE REVIEW

**Comment by mafatekhov** [comment]
@NatalieAntoshka 
line 149 - there is no method get in WordStorage
lines 102, 113 - log is not defined here

consider the following scenario:
```python
sentences = split_by_sentence(REFERENCE_TEXT)  # ref text is first 500 lines from original text
print(sentences)
ws = WordStorage()
for sentence in sentences:
    ws.from_corpus(tuple(sentence))
ngram = NGramTrie(2)
# for sentence in sentences:
encoded = encode(ws, sentences)

print(encoded)
for enc in encoded:
    ngram.fill_from_sentence(tuple(enc))
ngram.calculate_log_probabilities()
word1 = ws.get_id_of('she')
word2 = ws.get_id_of('will')
word3 = ws.get_id_of('not')
words = ngram.predict_next_sentence((word1,))
for word in words:
    print(ws.get_original_by(word))
```

**Comment by NatalieAntoshka** [comment]
Большое спасибо ! Вижу.
А какая будет оценка если оставить все без изменений ? 



Отправлено из мобильной Почты Mail.Ru


среда, 28 ноября 2018 г., 13:49 +0300 от notifications@github.com  <notifications@github.com>:

**Comment by mafatekhov** [comment]
@NatalieAntoshka bi-gram tests are not passing, it's 5 currently

**Comment by mafatekhov** [comment]
@NatalieAntoshka fix get in encode function:
```python
 for word in sentence:
            number = storage_instance.get(word)
            id_sent.append(number)
```

There is no such method in WordStorage class

File: lab_3/main.py [hunk 1]
```diff
@@ -140,16 +140,12 @@ def predict_next_sentence(self, prefix: tuple) -> list:
...
+    encode = []
     for sentence in corpus:
-        for word in sentence:
-            number = storage_instance.get(word)
-            id_sent.append(number)
-        corp_1.append(id_sent)
-        id_sent = []
-        continue
-    return corp_1
+        for key,value in storage_instance.items():
```
Comment (mafatekhov) [inline lab_3/main.py]: @NatalieAntoshka for some reason all of you ignore provided scenario of using your lab. We are doing this lab to be able to generate some sentences/text and to be able to use it not just to enable provided unit-tests. On a given scenario this still does not work. Correct implementation looks like this:
```python
def encode(storage_instance, corpus) -> list:
    encode = []
    for sentence in corpus:
        enc_sent = []
        for word in sentence:
            enc_sent.append(storage_instance.get_id_of(word))
        encode.append(enc_sent)
    return encode
```

**Pull request closed without merging**